src/main/java/org/elasticsearch/index/query/TemplateQueryBuilder.java

```diff
@@ -37,6 +46,27 @@
 
     private ScriptService.ScriptType templateType;
 
+    /** Name to reference this type of query. */
+    public static final String NAME = "template";
+    /** Name of query parameter containing the template string. */
+    public static final String QUERY = "query";
+    /** Name of query parameter containing the template parameters. */
+    public static final String PARAMS = "params";
+
+    private final ScriptService scriptService;
+
+    private final static Map<String,ScriptService.ScriptType> parametersToTypes = new HashMap<>();
+    static {
+        parametersToTypes.put("query", ScriptService.ScriptType.INLINE);
+        parametersToTypes.put("file", ScriptService.ScriptType.FILE);
+        parametersToTypes.put("id", ScriptService.ScriptType.INDEXED);
+    }
+
+    @Inject
+    public TemplateQueryBuilder(ScriptService scriptService) {
+        this.scriptService = scriptService;
+    }
```

Comment:
We should have access to the services on the coordinating node, and they should be usable there, so I think this will come out in further refactoring.